htlcswitch/link.go

```diff
@@ -766,6 +791,26 @@ func (l *channelLink) syncChanStates() error {
 			}
 		}
 
+		// Before we process the ChanSync message, if this is a taproot
+		// channel, then we'll init our musig2 nonces state.
+		if chanState.ChanType.IsTaproot() {
+			l.log.Infof("initializing musig2 nonces")
+
+			if remoteChanSyncMsg.LocalNonce == nil {
+				return fmt.Errorf("remote nonce is nil")
+			}
+
+			syncMsg := remoteChanSyncMsg
+			remoteNonce := &musig2.Nonces{
+				PubNonce: *syncMsg.LocalNonce,
+			}
+			err := l.channel.InitRemoteMusigNonces(remoteNonce)
```

Comment:
Let's say that the peer doesn't use the same shachain nonce scheme and instead sends a new nonce in `channel_reestablish`. If we owe them a `commitment_signed`, we _should_ be using this new nonce or else the peer will reject the update. However, that means that we're using our signing nonce with a new nonce to sign the same message and I'm not sure if that's ok (the peer could be lying and may have actually received the `commitment_signed` previously)?

Reply:
IIUC, at this point, the `funding_locked` has already been processed. 

> If we owe them a commitment_signed, we should be using this new nonce or else the peer will reject the update.

Since we init the nonces anew here, our `commit_sig` message will use the _new_ nonce. 

The signing nonce (the one we generate right when we want to sign a new state) is always generated anew. This doesn't change for when we need to send a new signature with channel reest. I modified the chan sync tests in 6, and can add another assertion there to ensure that a new nonce is always used there.